src/main/java/pl/rejmanbeata/bakery/controller/ClientController.java

```diff
@@ -0,0 +1,69 @@
+package pl.rejmanbeata.bakery.controller;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import pl.rejmanbeata.bakery.model.client.Client;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/clients")
+public class ClientController {
+    private final List<Client> clientList = new ArrayList<>();
+
+    @GetMapping("/{name}")
+    public ResponseEntity<Client> getClientByName(@PathVariable String name) {
+        Optional<Client> client = clientList.stream()
+                .filter(cli -> cli.getName().equalsIgnoreCase(name))
+                .findFirst();
+
+        return client
+                .map(value -> new ResponseEntity<>(value, HttpStatus.OK))
+                .orElseGet(() -> new ResponseEntity<>(HttpStatus.NOT_FOUND));
+    }
+
+    @PostMapping
+    public ResponseEntity<Client> createClient(@RequestBody Client client) {
+        clientList.add(client);
+        return new ResponseEntity<>(client, HttpStatus.CREATED);
+    }
+
+    @PutMapping("/{name}")
+    public ResponseEntity<Client> updateClient(@PathVariable String name, @RequestBody Client updatedClient) {
+        Optional<Client> existingClient = clientList.stream()
+                .filter(cli -> cli.getName().equalsIgnoreCase(name))
```

Comment:
please do not use short variables' name as `cli` ;) Let's use full name `client`, or shortened `c` (but I'd vote for `client` actually because it says more)